index.bs

```diff
@@ -450,8 +450,27 @@ To <dfn>pullDatagrams</dfn>, given a {{WebTransport}} object |transport|, run th
   1. Set |datagrams|.{{[[IncomingDatagramsPullPromise]]}} to a new promise.
   1. Return |datagrams|.{{[[IncomingDatagramsPullPromise]]}}.
 1. Let |bytes| and |timestamp| be the result of [=dequeuing=] |queue|.
-1. Let |chunk| be a new {{Uint8Array}} object representing |bytes|.
-1. [=ReadableStream/Enqueue=] |chunk| to |transport|.{{[[Datagrams]]}}.{{WebTransportDatagramDuplexStream/[[Readable]]}}.
+1. Let |buffer|, |offset|, |maxBytes|, and |chunk| be null.
+1. If |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}'s
+   [=ReadableStream/current BYOB request view=] is not null, then:
+  1. Let |view| be |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}'s
+     [=ReadableStream/current BYOB request view=].
+  1. If |view|'s [=BufferSource/byte length=] is less than the size of |bytes|,
+     [=ReadableStream/error=] |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}
+     with a {{RangeError}}.
+  1. Otherwise:
```

Comment:
This needs to bail, otherwise the algorithm proceeds to "Write |bytes| into" `null` below. Unlike with _throw_, which implicitly does this, I don't think we can assume abort to be implicit here, at least it seems unclear to me, especially with the "Otherwise" directly following this statement.

I suggest something like, "and abort all steps", since we're technically inside nested steps here.

This should let us remove this "Otherwise" as well, and help flatten things.